### PR TITLE
CompactFormatter precision and surrogate pairs #528

### PR DESCRIPTION
**Pull request by jmehrens**
This is a fix for https://github.com/eclipse-ee4j/mail/issues/528.

The code coverage has been improved.  The CompactFormatter is at 99.0% and the test is at 97.7%.
Test was modified to clean up some of the SpotBugs warnings.

When this is approved I'll backport the changed to 1.x.

### CODE REVIEW

File: mail/src/main/java/com/sun/mail/util/logging/CompactFormatter.java
Comment (jmehrens) [inline mail/src/main/java/com/sun/mail/util/logging/CompactFormatter.java:481]: FYI this is dead code caught by code coverage.  It seems from the API docs that StackTraceElement::getClassName is non null.  Other parts of the code already treat it as non-null.

File: doc/release/CHANGES.txt
Comment (lukasj) [inline doc/release/CHANGES.txt:34]: this needs to go to 2.0.2 version. 2.0.1 is on review for almost 3 weeks already , see https://github.com/jakartaee/specifications/pull/339

**Review by lukasj — APPROVED** [review]
LGTM. I don't want to merge this before 2.0.1 gets approved and out, so will do that once that happens (I have no impact on the approval itself, I can only wait).

Wrt backport to v1.x - I plan to release 1.6.7 early next week to fix awful bug there, so if you have something else you want to have there, just create PR for it. Thanks!

File: doc/release/CHANGES.txt
Comment (jmehrens) [reply doc/release/CHANGES.txt]: Added a new section for 2.0.2 release to the changes.txt.

**Pull request merged**
